l2-contracts/contracts/AttesterRegistry.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+contract AttesterRegistry {
+    // Owner of the contract (the Authority contract).
+    address public owner;
+
+    // A map of indexes => attesters (used for efficient lookups).
+    mapping(uint256 => Attester) public attesters;
+    // Array to keep track of attester indexes (used for iterations).
+    uint256[] public attesterIndexes;
+    // The last unique value used as an attester index (used for supporting arbitrary removals).
+    uint256 public attesterSequence;
+
+    // The current committee list. Weight and public key are stored explicitly
+    // since they might change after committee selection.
+    CommitteeAttester[] public committee;
+    // The committee list for the next epoch.
+    CommitteeAttester[] public nextCommittee;
+
+    // The current epoch number.
+    uint256 public epoch;
+    // The number of epochs that an attester must be inactive before being
+    // possible to remove it. Needs to be at least 2, in order to guarantee
+    // that any attester in the current or next committee lists is still
+    // available in the registry.
+    uint256 public constant INACTIVITY_DELAY = 2;
+
+    struct Attester {
+        // Voting weight.
+        uint256 weight;
+        // ECDSA public key.
+        bytes pubKey;
```

Comment:
Why isn't this an address?

Reply:
`address` would be a truncated 20 bytes of the pub key hash. I assumed the pub key itself is needed here for data availability. (although it can be retrieved off-chain from a signature and get hashed+truncated to be compared to the address.)